### PR TITLE
wasi: env initialization fix

### LINKED ISSUE #261: Bug in WASISnapshotPreview1WithConfig for Environment variables
I found a small bug in the newly added Environment support in WASI:

```go
if len(c.Environ) > 0 {
  environ := make([]string, len(c.Environ))
  for k, v := range c.Environ {
    environ = append(environ, fmt.Sprintf("%s=%s", k, v))
  }
…
```
with this logic, `environ` always contains `""` as a first item and it fails validation.

### PR DESCRIPTION
**Pull request by pims**
Closes #261

### CODE REVIEW

File: wasi.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func WASISnapshotPreview1WithConfig(c *WASIConfig) *HostFunctions {
 		opts = append(opts, opt)
 	}
 	if len(c.Environ) > 0 {
-		environ := make([]string, len(c.Environ))
+		environ := make([]string, 0)
```
Comment (codefromthecrypt) [inline wasi.go:60]: Thanks and sorry about the bug!
```suggestion
		environ := make([]string, 0, len(c.Environ))
```

**Comment by codefromthecrypt** [comment]
Thanks for the help and sorry about the construction debris. We're trying to finish major api breaks this weekend.

**Comment by pims** [comment]
Nothing to be sorry about! Was about to contribute Environment support but happy to see someone did it before me!

**Comment by codefromthecrypt** [comment]
ps if you are using something besides tinygo and/or need a WASI API not defined here, please pipe up in the issue https://github.com/tetratelabs/wazero/issues/271

**Pull request merged**